File: sample-app/src/main/java/com/psoffritti/slidingpanel/sampleapp/examples/utils/adapters/ViewPagerAdapter.java
package com.psoffritti.slidingpanel.sampleapp.examples.utils.adapters;

import android.util.Pair;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class ViewPagerAdapter extends FragmentPagerAdapter {

    private Pair<Fragment, String>[] fragments;

    @SafeVarargs
    public ViewPagerAdapter(FragmentManager fragmentManager, Pair<Fragment, String>... fragments) {
        super(fragmentManager);
        this.fragments = fragments;
    }

    @Override
    public Fragment getItem(int position) {
        return fragments[position].first;
    }

    @Override
    public int getCount() {
        return fragments.length;
    }

    @Override
    public CharSequence getPageTitle (int position) {
        return fragments[position].second;
    }
}

File: sample-app/src/main/java/com/akbolatss/workshop/slidingpanel/sampleapp/utils/adapters/ViewPagerAdapter.kt
package com.akbolatss.workshop.slidingpanel.sampleapp.utils.adapters

import android.util.Pair
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class ViewPagerAdapter @SafeVarargs
constructor(
    fragmentManager: FragmentManager,
    private vararg val fragments: Pair<Fragment, String>
) :
    FragmentPagerAdapter(
        fragmentManager,
        BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT
    ) {

    override fun getItem(position: Int): Fragment {
        return fragments[position].first
    }

    override fun getCount(): Int {
        return fragments.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return fragments[position].second
    }
}